### PR TITLE
staticanalysis/bot: Update parsepatch to latest version and add a regression test for checking if an issue is in a revision

### LINKED ISSUE #1674: Issues detected in new files are not published
https://phabricator.services.mozilla.com/D11190 is adding a test_results.py with some issues. The issues are detected but not published because the bot thinks they are not in the patch:
```
{
  "time": 1541612296.424006,
  "revision": {
    "source": "phabricator",
    "diff_phid": "PHID-DIFF-ys2l54m2lvbwewiekczg",
    "phid": "PHID-DREV-ha523hqb5kuful6kliyz",
    "id": 11190,
    "url": "https://phabricator.services.mozilla.com/D11190",
    "has_clang_files": false,
    "title": "Bug 1503599 - [mozlint] Default lineno to 0 in result.Issue if unset",
    "bugzilla_id": "1503599"
  },
  "issues": [
    {
      "analyzer": "mozlint",
      "level": "error",
      "path": "python/mozlint/test/test_result.py",
      "linter": "flake8",
      "line": 8,
      "column": 1,
      "nb_lines": 1,
      "rule": "F401",
      "message": "'pytest' imported but unused",
      "validation": {
        "third_party": false,
        "disabled_rule": false
      },
      "in_patch": false,
      "is_new": false,
      "validates": true,
      "publishable": false
    },
    {
      "analyzer": "mozlint",
      "level": "error",
      "path": "python/mozlint/test/test_result.py",
      "linter": "flake8",
      "line": 16,
      "column": 25,
      "nb_lines": 1,
      "rule": "E711",
      "message": "comparison to None should be 'if cond is None:'",
      "validation": {
        "third_party": false,
        "disabled_rule": false
      },
      "in_patch": false,
      "is_new": false,
      "validates": true,
      "publishable": false
    }
  ],
  "patches": {}
}
```

### PR DESCRIPTION
**Pull request by marco-c**
Fixes #1674 and adds a regression test for it.

### CODE REVIEW

File: src/staticanalysis/bot/tests/test_revisions.py [hunk 1]
```diff
@@ -78,3 +78,81 @@ def test_clang_files(mock_revision):
...
+            return ''
+
+        def validates():
+            return True
+
+    issue_in_new_file = MyIssue('new.txt', 1)
+    issue_in_existing_file_touched_line = MyIssue('modified.txt', 3)
+    issue_in_existing_file_not_changed_line = MyIssue('modified.txt', 1)
+    issue_in_existing_file_added_line = MyIssue('added.txt', 4)
+    issue_in_not_changed_file = MyIssue('notexisting.txt', 1)
```
Comment (jankeromnes) [inline src/staticanalysis/bot/tests/test_revisions.py]: Over-nit:

```suggestion
    issue_in_unchanged_file = MyIssue('nonexistent.txt', 1)
```

File: src/staticanalysis/bot/tests/test_revisions.py [hunk 2]
```diff
@@ -78,3 +78,81 @@ def test_clang_files(mock_revision):
...
+
+        def as_text():
+            return ''
+
+        def validates():
+            return True
+
+    issue_in_new_file = MyIssue('new.txt', 1)
+    issue_in_existing_file_touched_line = MyIssue('modified.txt', 3)
+    issue_in_existing_file_not_changed_line = MyIssue('modified.txt', 1)
```
Comment (jankeromnes) [inline src/staticanalysis/bot/tests/test_revisions.py]: Over-nit:

```suggestion
    issue_in_existing_file_unchanged_line = MyIssue('modified.txt', 1)
```

File: src/staticanalysis/bot/tests/test_revisions.py [hunk 3]
```diff
@@ -78,3 +78,81 @@ def test_clang_files(mock_revision):
...
+    assert rev.lines['modified.txt'] == [3]
+    assert 'added.txt' in rev.lines
+    assert rev.lines['added.txt'] == [4]
+    assert 'new.txt' in rev.files
+    assert 'modified.txt' in rev.files
+    assert 'added.txt' in rev.files
+
+    assert rev.contains(issue_in_new_file)
+    assert rev.contains(issue_in_existing_file_touched_line)
+    assert not rev.contains(issue_in_existing_file_not_changed_line)
```
Comment (jankeromnes) [inline src/staticanalysis/bot/tests/test_revisions.py]:
```suggestion
    assert not rev.contains(issue_in_existing_file_unchanged_line)
```

File: src/staticanalysis/bot/tests/test_revisions.py [hunk 4]
```diff
@@ -78,3 +78,81 @@ def test_clang_files(mock_revision):
...
+    assert rev.contains(issue_in_existing_file_added_line)
+    assert not rev.contains(issue_in_not_changed_file)
```
Comment (jankeromnes) [inline src/staticanalysis/bot/tests/test_revisions.py]:
```suggestion
    assert not rev.contains(issue_in_unchanged_file)
```

**Comment by marco-c** [comment]
> Have you verified that this test correctly fails with the older parsepatch, and correctly passes with the newer parsepatch?

Yes, I wrote the test before the update and it was failing. Then it started to pass after I updated it.

**Pull request merged**